src/Console/Output/ErrorOutput.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Console\Output;
+
+use PhpCsFixer\Error\Error;
+use PhpCsFixer\Linter\LintingException;
+use Symfony\Component\Console\Formatter\OutputFormatter;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @author SpacePossum
+ *
+ * @internal
+ */
+final class ErrorOutput
+{
+    /**
+     * @var OutputInterface
+     */
+    private $output;
+
+    /**
+     * @var bool
+     */
+    private $isDecorated;
+
+    public function __construct(OutputInterface $output)
+    {
+        $this->output = $output;
+        $this->isDecorated = $output->isDecorated();
+    }
+
+    /**
+     * @param string  $process
+     * @param Error[] $errors
+     */
+    public function listErrors($process, array $errors)
+    {
+        $this->output->writeln(array('', sprintf(
+            'Files that were not fixed due to errors reported during %s:',
+            $process
+        )));
+
+        $showDetails = $this->output->getVerbosity() >= OutputInterface::VERBOSITY_VERBOSE;
+        $showTrace = $this->output->getVerbosity() >= OutputInterface::VERBOSITY_VERY_VERBOSE;
+        foreach ($errors as $i => $error) {
+            $this->output->writeln(sprintf('%4d) %s', $i + 1, $error->getFilePath()));
+            if ($showDetails && null !== $e = $error->getSource()) {
+                $this->output->writeln(array(
+                    '      '.($this->isDecorated ? '<bg=yellow;fg=black;>Details</>' : 'Details'),
```

Comment:
thanks for checking, will update!
I think there are more major projects that would benefit from that optimization :)